js/github.js

```diff
@@ -9,10 +10,38 @@ const doesGitHubCookieExist = () => {
   return !!githubCookieStr;
 };
 
+const SIGN_IN_GITHUB_OAUTH = () => {
+  let originURL = window?.location?.origin ?? '/';
+  let githubOauth =
+    'https://github.com/login/oauth/authorize?client_id=23c78f66ab7964e5ef97&state=';
+  let encodedOriginURL = btoa(originURL);
+
+  return githubOauth + encodedOriginURL;
+};
+
 const updateGitHubLink = () => {
+  const githubMock = JSON.parse(localStorage.getItem('githubMock')) ?? {};
+  const githubMockExpired = new Date().getTime() > (githubMock.expiresIn ?? 0);
+
+  let signUpLink = GITHUB_MOCK_URL;
+
+  if (githubMock.visited && !githubMockExpired) {
+    signUpLink = SIGN_IN_GITHUB_OAUTH();
+  }
+
   const allLoginBtns = document.querySelectorAll('.btn-login');
   allLoginBtns.forEach((btn) => {
-    btn.setAttribute('href', 'https://github.realdevsquad.com');
+    btn.setAttribute('href', signUpLink);
+
+    btn.addEventListener('click', (event) => {
+      event.preventDefault();
+      let githubMock = {
```

Comment:
Can it be made `const` ?

Reply:
are we planning to export it?

Reply:
Nope I just wanted the variable to be a constant, so that it can't be reassigned